.github/workflows/build-workflow.yml

```diff
@@ -3,15 +3,26 @@ name: Slides Workflow
 on:
   push:
 
+env:
+  ## Override default value for Docker cached image
+  IMAGE_CACHE_NAME: "dduportal/slides:cicd-lectures"
+
 jobs:
   build:
     name: 'Build Slides'
     runs-on: ubuntu-latest
     steps:
       - name: 'Checkout'
         uses: actions/checkout@master
+      - name: 'Login to DockerHub'
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
       - name: 'Build'
         run: make build
+      - name: 'Caching for later builds'
+        run: docker push "${IMAGE_CACHE_NAME}"
```

Comment:
This might be farfetched so feel free to ignore this, but isn't there any risk of side effects pushing the `latest` image on every build?

Let's say ongoing work, that updates build environment and make breaking changes inside it. It will get pushed and break any other builds?

Reply:
if you have a PR which "breaks" the expected behavior, and then pushed its cache to the repo,
then the other branches would invalidate the cache and build it (and then override it),
because the context would be changed (breaking the image means changing the Dockerfile, the package.json, or any element of the context).


However we could play safely by only pushing on master branch, WDYT? In this case, PR only on the content (excluded from the context) would still have cache hit, but PRs based on changing tooling would invaidate the cache, and not pushing until merged.

Reply:
Change applied, let me know :)

Reply:
Thanks 🎉